src/realm/realmReducer.js

```diff
@@ -35,7 +35,6 @@ export default (state: RealmState = initialState, action: Action): RealmState =>
 
     case REALM_INIT: {
       return {
-        ...state,
```

Comment:
Since this involves spreading a value and type-checking in a reducer, I was vaguely reminded of [this Flow bug](https://github.com/zulip/zulip-mobile/pull/4046#discussion_r412562505).

I verified empirically that it isn't in play after this commit. (Making a property `undefined` trips Flow.) The next commit helps verify that everything's working as expected by reading the code. So in this case it wasn't necessary to go re-digest what exactly that Flow bug is.

Reply:
On re-reading the Flow issue Ray linked to, it sounds like maybe the bug was happening *before* this commit? Anyway, the important thing is that it's not happening after it. 😆

Reply:
This PR looks good to me, but Ray may have more to say.